From 1d85f2af314786b2d4cf8d4564c3b8b565c577e6 Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Thu, 27 Jul 2023 14:34:43 +0200
Subject: [PATCH 08/16] media: stm32: dcmi: continuous mode capture in JPEG

The performances of the DMA-MDMA chaining allows to rely
on the continuous capture mode for JPEG as well, allowing
to achieve a faster framerate in all resolutions.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/linux.git 04e0ed75d705d54de71132157bbeb765f6e70d25]
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Change-Id: I9f5dfe40d63f08e242470c1c0c3b39dbb3846ac4
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/320418
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
---
 drivers/media/platform/st/stm32/stm32-dcmi.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/media/platform/st/stm32/stm32-dcmi.c b/drivers/media/platform/st/stm32/stm32-dcmi.c
index 0834bbb3343a3..5763facdfa5e6 100644
--- a/drivers/media/platform/st/stm32/stm32-dcmi.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmi.c
@@ -836,10 +836,6 @@ static int dcmi_start_streaming(struct vb2_queue *vq, unsigned int count)
 	if (dcmi->do_crop)
 		dcmi_set_crop(dcmi);
 
-	/* Enable jpeg capture */
-	if (dcmi->sd_format->fourcc == V4L2_PIX_FMT_JPEG)
-		reg_set(dcmi->regs, DCMI_CR, CR_CM);/* Snapshot mode */
-
 	/* Enable dcmi */
 	reg_set(dcmi->regs, DCMI_CR, CR_ENABLE);
 
-- 
2.43.0

